LambdaEngine/Source/Resources/ResourceManager.cpp

```diff
@@ -778,7 +886,213 @@ namespace LambdaEngine
 
 		pMaterial->pAOMetallicRoughnessMap		= pCombinedMaterialTexture;
 		pMaterial->pAOMetallicRoughnessMapView	= pCombinedMaterialTextureView;
-		RegisterLoadedTextureWithView(pCombinedMaterialTexture, pCombinedMaterialTextureView);
+		return RegisterLoadedTextureWithView(pCombinedMaterialTexture, pCombinedMaterialTextureView);
+	}
+
+	bool ResourceManager::UnloadMesh(GUID_Lambda guid)
+	{
+		auto meshIt = s_Meshes.find(guid);
+		if (meshIt != s_Meshes.end())
+		{
+			SAFEDELETE(meshIt->second);
+			s_Meshes.erase(meshIt);
+
+			auto meshGUIDToNameIt = s_MeshGUIDsToNames.find(guid);
+
+			if (meshGUIDToNameIt != s_MeshGUIDsToNames.end())
+			{
+				//Clean Mesh Name -> GUID
+				auto meshNameToGUIDIt = s_MeshNamesToGUIDs.find(meshGUIDToNameIt->second);
+				if (meshNameToGUIDIt != s_MeshNamesToGUIDs.end()) s_MeshNamesToGUIDs.erase(meshNameToGUIDIt);
+
+				auto animationsIt = s_MeshNamesToAnimationGUIDs.find(meshGUIDToNameIt->second);
+
+				if (animationsIt != s_MeshNamesToAnimationGUIDs.end())
+				{
+					for (GUID_Lambda animationGUID : animationsIt->second)
+					{
+						UnloadAnimation(animationGUID);
+					}
+
+					//Clean Mesh Name -> Animation GUID
+					s_MeshNamesToAnimationGUIDs.erase(animationsIt);
+				}
+
+				//Clean Mesh GUID -> Name
+				s_MeshGUIDsToNames.erase(meshGUIDToNameIt);
+			}
+
+			return true;
+		}
+
+		LOG_ERROR("[ResourceManager]: Failed to unload Mesh: %d", guid);
+		return false;
+	}
+
+	bool ResourceManager::UnloadMaterial(GUID_Lambda guid)
```

Comment:
This function and all unload functions below could return true even if eg. the GUID -> name mapping is broken. To fix this, you could replace the nested scopes with:

`
if (myIt == mapping.end())
{
        return false;
}
`